### ISSUE TITLE
Hardcoded Message Attributes MD5 checksum causes error in AWS Java SDK

### ISSUE THREAD

**Issue by awesometown** [spulec/moto#918]
I've been trying to get set up to use Atlassian's localstack for local development of a Java application. When trying to send SQS messages that use MessageAttributes I receive errors from the SDK that the returned MD5 checksum does not match what was expected:

```
com.amazonaws.AmazonClientException: MD5 returned by SQS does not match the calculation on the original request. (MD5 calculated by the message attributes: "235c5c510d26fb653d073faed50ae77c", MD5 checksum returned: "324758f82d026ac6ec5b31a3b192d1e3")
```

In the [issue I filed with localstack](https://github.com/atlassian/localstack/issues/78), one of the maintainers noted that moto is using a [hard-coded MD5 checksum](https://github.com/spulec/moto/blob/896f040fca975865c1d441d90016446c72db3ed4/moto/sqs/responses.py#L343).

Would it be possible for moto to calculate and return the expected checksum value?

**Comment by JackDanger** [comment]
I've implemented the fix for this in #919. Thanks for reporting it, @awesometown! Client software shouldn't break just because it enforces good security practices.

**Comment by awesometown** [comment]
Sorry for the delayed acknowledgement, but thanks for looking into this :)
